### DISCUSSION TITLE
Wix Toolset 3.11.2 extension fails to install on Visual Studio 2022 due to System.TypeLoadException

### DISCUSSION THREAD

**Discussion by daniel-tran** [wixtoolset/issues#7741] [Questions]
_Note: This was originally posted on the Microsoft Visual Studio Developer Community Forum (see [here](https://developercommunity.visualstudio.com/t/Wix-Toolset-3112-extension-fails-to-in/10470174)) and the general advice provided was to relay the issue here._

I’m trying to install the Wix v3 Toolset for Visual Studio 2022 extension but it’s consistently failing due to the following exception:

```
System.TypeLoadException: Could not find Windows Runtime type 'Windows.Networking.Connectivity.NetworkCostType'
```

Other details:
- **Visual Studio version**: Microsoft Visual Studio Professional 2022 (17.7.4)
- **Windows OS**: Microsoft Windows Server 2016 Standard (10.0.14393 N/A Build 14393) - Note that this is a Server Core installation
- [This is the log file from the failed installation](https://aka.ms/dc/file?name=B158898a8bc7040f3aa01369d6acf3a18638307607133613512_Wix2022Failure.log&tid=158898a8bc7040f3aa01369d6acf3a18638307607133613512)

This happens when running the .vsix directly as well as through the Manage Extensions tab in Visual Studio 2022. Rebooting the machine and repairing Visual Studio 2022 also doesn’t seem to change the result.
The initial reply from the Microsoft Visual Studio Developer Community Forum was that the error is due to some missing DLL, although what specific DLL is needed is still currently unknown (although I am speculating is might be "Windows.Networking.Connectivity.dll" based on the exception message).

Any guidance on what to do about this would be great.

**Comment by chrpai** [comment]
What workloads and components are installed?  I'm not sure it's ever been 100% exactly documented but I suspect if you install desktop development workload it'll probably resolve it.

**Reply by daniel-tran** [reply]
In terms of desktop development workloads, I have the C++, NET and Universal Windows Platform workloads installed. I've also tried re-installing them but that doesn't appear to change the result.
